tests/test_indexes.py

```diff
@@ -1038,11 +1039,12 @@ def test_timedelta_div() -> None:
     check(assert_type(index // [1], pd.TimedeltaIndex), pd.TimedeltaIndex, pd.Timedelta)
 
     check(assert_type(delta / index, "pd.Index[float]"), pd.Index, float)
-    # ZeroDivisionError
-    # check(assert_type([delta] / index, "pd.Index[float]"), pd.Index, float)
+    check(assert_type([delta] / index, "pd.Index[float]"), pd.Index, float)
     check(assert_type(delta // index, "pd.Index[int]"), pd.Index, np.longlong)
-    # ZeroDivisionError
-    # check(assert_type([delta] // index, "pd.Index[int]"), pd.Index, np.longlong)
+    dtype: type[np.integer] = np.int64
+    if sys.platform == "win32":
+        dtype = np.int32
+    check(assert_type([delta] // index, "pd.Index[int]"), pd.Index, dtype)
```

Comment:
I think you can use `np.intp` and it will be platform independent.  Did you try that?

Reply:
Yes, it failed. See https://github.com/pandas-dev/pandas-stubs/actions/runs/5879471018/job/15943590665

Reply:
Can you try `np.integer` ? Both `isinstance(np.int32(8), np.integer)` and `isinstance(np.int64(10), np.integer)` are `True` for me on Windows and Linux.

Reply:
Thank you, using np.signedinteger worked as well :)